pkg/controller/datavolume/import-controller_test.go

```diff
@@ -1466,6 +1466,49 @@ var _ = Describe("All DataVolume Tests", func() {
 			Expect(readyCondition.Message).To(Equal(""))
 		})
 
+		It("Should not use PVC to update status if PVC is marked for deletion", func() {
+			reconciler = createImportReconciler(NewImportDataVolume("test-dv"))
+			_, err := reconciler.Reconcile(context.TODO(), reconcile.Request{NamespacedName: types.NamespacedName{Name: "test-dv", Namespace: metav1.NamespaceDefault}})
+			Expect(err).ToNot(HaveOccurred())
+
+			pvc := &corev1.PersistentVolumeClaim{}
+			err = reconciler.client.Get(context.TODO(), types.NamespacedName{Name: "test-dv", Namespace: metav1.NamespaceDefault}, pvc)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(pvc.Name).To(Equal("test-dv"))
+
+			By("Setting PVC to Succeeded phase and marking for deletion")
+			pvc.SetAnnotations(make(map[string]string))
+			pvc.GetAnnotations()[AnnPodPhase] = string(corev1.PodSucceeded)
+
+			// Set finalizer so PVC is not removed when Delete is called
+			pvc.Finalizers = append(pvc.Finalizers, "test-finalizer")
+			err = reconciler.client.Update(context.TODO(), pvc)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Mark PVC for deletion
+			err = reconciler.client.Delete(context.TODO(), pvc)
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
You don't have to use the client to setup the state, you can just pass the objects in the target state to `createImportReconciler`